_profiles/drupal/resource-versioning/index.md

```diff
@@ -0,0 +1,179 @@
+  resource object may afford the capability of requesting a resource object
+  as it existed in a prior or successive state. This capability is herein
+  defined as _resource versioning_.
+  
+  This profile establishes a protocol for resource versioning by defining a
+  query parameter and its semantics in order to identify arbitrary revisions
+  of a resource and how a server should interpret this query parameter.
+
+minimum_jsonapi_version: 1.0
+minimum_jsonapi_version_explanation:
+
+discussion_url: https://www.drupal.org/project/issues/jsonapi
+
+editors:
+  - name: Gabriel Sullice
+    email: gabriel@sullice.com
+  - name: Mateu Aguiló Bosch
+    email: mateu.aguilo.bosch@gmail.com
+    website: https://humanbits.es
+  - name: Wim Leers
+    email: work@wimleers.com
+    website: https://wimleers.com
+
+categories:
+  - Resource Versioning
+---
+  
+# Concepts
+Resources on a server may undergo changes and the state of a resource with an
+arbitrary number of changes may be accessible. It is often useful to retrieve
+resources as they existed at the time of their creation or in various states of
+change for editorial or archival purposes. This profile establishes a protocol
+for accessing resources in those various states.
+
+For the purpose of understanding this profile, it is helpful to establish a
+vocabulary for describing various possible states of a resource.
+
+A _revision_ is to be understood as an identifiable state of a resource after
+its creation or after some number of changes.
+
+A _version_ is to be understood as a revision of a resource that is available,
+or was previously available, without any version negotiation. In other words, as
+a revision that is or was the default revision of a resource.
+
+A _working copy_ is the revision to which new changes can be made or to which
+they will be applied. Colloquially, a working copy is often thought of as the
+"tip" of a version history.
+
+For example, a resource which existed as a draft and underwent multiple changes
+before the resource was published may have many revisions. Only the published
+revision would be considered a version. The revisions prior to first version
+would have been known as the working copy as each one was created.
+
+If the version is then checked out for further changes (which may or may not be
+published) each new revision becomes the working copy.
+
+Sophisticated servers may support multiple versions of a resource as well as
+multiple working copies of a resource (e.g., to support multiple
+languages or within a version control system which supports multiple branches).
+
+This profile creates a standard for JSON:API to support these diverse versioning
+schemes.
+# Query Parameter
+
+## Usage
+
+An endpoint **MAY** support a `resource_version` query parameter to allow a
+client to indicate which version(s) of a resource should be returned.
+
+If an endpoint does not support the `resource_version` parameter, it **MUST**
```

Comment:
Now that JSON:API is recommending camel casing, it might be better to call this `resourceVersion`, to match the format most likely to be used by other profiles.

Reply:
Right... hm. Our implementation will need to change, but that's okay, it hasn't landed yet.

Reply:
c351d9ceee9327bf10e6d48417b90a1792bab44d